### PR TITLE
Getting ready for the next release!

### PR DESCRIPTION
**Pull request by alexreardon**
- upgrading all dev deps
- including type check files in typecheck
- moving dev dep of typescript to typescript@5

### CODE REVIEW

**Comment by alexreardon** [comment]
cc @Andarist

File: package.json [hunk 1]
```diff
@@ -45,16 +45,17 @@
...
+    "jest": "^29.5.0",
+    "jest-environment-jsdom": "^29.5.0",
+    "prettier": "^2.8.8",
+    "rimraf": "^5.0.0",
+    "size-limit": "^8.2.4",
     "ts-expect": "^1.3.0",
-    "ts-jest": "^27.1.3",
-    "typescript": "^4.6.4"
+    "ts-jest": "^29.1.0",
+    "typescript": "^5.0.4"
```
Comment (alexreardon) [inline package.json:58]: Using TypeScript@5 locally now

**Comment by alexreardon** [comment]
### Question: should we make our next release a `major`?

- types and exports still match current version (`2.1.1`)
- consumers on TS@4 will stop getting autocomplete for event names

Right now I slightly lean towards a `major`, but keen to hear what others think

File: package.json (hunk 1 above)
Comment (alexreardon) [reply package.json:59]: Drawback being that we are no longer testing on TS@4. I could add some logic to test for TS@4, but I don't think it is worth the effort - especially if we release a `major`

Comment (Andarist) [reply package.json:59]: > I could add some logic to test for TS@4, but I don't think it is worth the effort - especially if we release a major

Agreed, especially that this only affects completions etc. Typechecking itself should be unaffected and we don't have a good way to test completions/errors/type displays/etc.

**Comment by Andarist** [comment]
> consumers on TS@4 will stop getting autocomplete for event names

Did you test this out? The intention was to keep the completions *where* they were already available - but `bindAll` never had any completions for those event names and that has been improved with my changed *but* only for 5.0+. If anything has been broken with the existing completions, I could investigate this but a repro case for this would be appreciated :)

> types and exports still match current version (2.1.1)

`bindAll`'s types has changed the questions is if you consider that to be part of the public API bound by semver or not.

**Pull request merged**